android/test_app/app/src/main/java/org/pytorch/testapp/MainActivity.java

```diff
@@ -0,0 +1,158 @@
+                if (mBackgroundHandler != null) {
+                  mBackgroundHandler.post(mModuleForwardRunnable);
+                }
+              });
+        }
+      };
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+    mTextView = findViewById(R.id.text);
+    startBackgroundThread();
+    mBackgroundHandler.post(mModuleForwardRunnable);
+  }
+
+  protected void startBackgroundThread() {
+    mBackgroundThread = new HandlerThread(TAG + "_bg");
+    mBackgroundThread.start();
+    mBackgroundHandler = new Handler(mBackgroundThread.getLooper());
+  }
+
+  @Override
+  protected void onDestroy() {
+    stopBackgroundThread();
+    super.onDestroy();
+  }
+
+  protected void stopBackgroundThread() {
+    mBackgroundThread.quitSafely();
+    try {
+      mBackgroundThread.join();
+      mBackgroundThread = null;
+      mBackgroundHandler = null;
+    } catch (InterruptedException e) {
+      Log.e(TAG, "Error stopping background thread", e);
+    }
+  }
+
+  @WorkerThread
+  @Nullable
+  protected Result doModuleForward() {
+    if (mModule == null) {
+      final long[] shape = BuildConfig.INPUT_TENSOR_SHAPE;
+      long numElements = 1;
+      for (int i = 0; i < shape.length; i++) {
+        numElements *= shape[i];
+      }
+      mInputTensorBuffer = Tensor.allocateFloatBuffer((int) numElements);
+      mInputTensor = Tensor.fromBlob(mInputTensorBuffer, BuildConfig.INPUT_TENSOR_SHAPE);
+      PyTorchAndroid.setNumThreads(1);
+      mModule = PyTorchAndroid.loadModuleFromAsset(getAssets(), BuildConfig.MODULE_ASSET_NAME);
+    }
+
+    final long startTime = SystemClock.elapsedRealtime();
+    final long moduleForwardStartTime = SystemClock.elapsedRealtime();
+    final IValue outputTuple = mModule.forward(IValue.listFrom(mInputTensor));
+    final IValue[] outputArray = outputTuple.toTuple();
+    final IValue out0 = outputArray[0];
+    final Map<String, IValue> map = out0.toDictStringKey();
+    if (map.containsKey("boxes")) {
+      final Tensor boxes = map.get("boxes").toTensor();
+      final Tensor scores = map.get("scores").toTensor();
+      final float[] boxesData = boxes.getDataAsFloatArray();
+      final float[] scoresData = scores.getDataAsFloatArray();
+      final int n = scoresData.length;
+      for (int i = 0; i < n; i++) {
+          android.util.Log.i(TAG,
+              String.format("Forward result %d: score %f box:(%f, %f, %f, %f)",
+                scoresData[i],
+                boxesData[4 * i + 0],
+                boxesData[4 * i + 1],
+                boxesData[4 * i + 2],
+                boxesData[4 * i + 3]));
+      }
+    } else {
+      android.util.Log.i(TAG, "Forward result empty");
+    }
+
+    final long moduleForwardDuration = SystemClock.elapsedRealtime() - moduleForwardStartTime;
+    final long analysisDuration = SystemClock.elapsedRealtime() - startTime;
```

Comment:
@IvanKobzarev @fmassa Why do we use two duration vars initialized and estimated on exactly the same point? This seems unnecessary. Perhaps the intention was for `moduleForwardDuration` to estimate the duration of the `mModule.forward()` call? Or perhaps the intention of `analysisDuration` is to measure just the unpacking of the response?